hnn_core/batch_simulate.py

```diff
@@ -76,20 +128,63 @@ def run(self, param_grid, return_output=True, combinations=True, n_jobs=1):
             Default is True.
         n_jobs : int, optional
             Number of parallel jobs. Default is 1.
+        backend : str or joblib.parallel.ParallelBackendBase instance, optional
+            The parallel backend to use. Can be one of `loky`, `threading`,
+            `multiprocessing`, or `dask`. Default is `loky`
+        verbose : int, optional
+            The verbosity level for parallel execution. Default is 50.
+        clear_cache : bool, optional
+            Whether to clear the results cache after saving each batch.
+            Default is False
 
         Returns
         -------
         results : list
             List of simulation results if return_output is True.
         """
+        _validate_type(param_grid, types=(dict,), item_name='param_grid')
+        _validate_type(n_jobs, types='int', item_name='n_jobs')
+        _check_option('backend', backend, ['loky', 'threading',
+                                           'multiprocessing', 'dask'])
+        _validate_type(verbose, types='int', item_name='verbose')
+
         param_combinations = self._generate_param_combinations(
```

Comment:
There should be a way to save parameters along with the simulation results using the `parameters_{start_idx}-{end_idx}` naming scheme

Reply:
Let's save this for the next PR

Reply:
In that PR we should add an option to specify which pieces of the simulation we'd like to export. The main objects of interest are
- Dipole waveform (already done)
- Neuron spiking `net.cell_response.spike_times` and `spike_types` and `spike_gids`
- Local field potentials through `net.extracellular_array`
- Voltages, currents, and calcium concentrations through `net.cell_response.vsec`, `isec`, and `ca`